.pipelines/android-ci.yml

```diff
@@ -0,0 +1,86 @@
+name: $(Year:yy).$(Month).$(DayOfMonth).$(rev:r)
+
+pr:
+  branches:
+    include:
+    - main
+    - feature/*
+    - release/*
+
+trigger:
+  branches:
+    include:
+    - main
+    - feature/*
+    - release/*
+  batch: True
+
+pool:
+  name: Azure Pipelines
+  vmImage: internal-macos-11
+
+steps:
+- task: Gradle@2
+  displayName: gradlew build
+  inputs:
+    wrapperScript: source/android/gradlew
+    cwd: source/android
+    publishJUnitResults: false
+  branches:
+    include:
+    - main
+    - feature/*
+    - release/*
+
+- task: Bash@3
+  displayName: Get NDK (won't be needed w/ Android Gradle Plugin 4.1+)
+  inputs:
+    targetType: inline
+    script: >-
+      #!/usr/bin/env bash
+
+      # Starting from AGP 4.1+, the required NDK version will be retrieved automatically, 
+      unset ANDROID_NDK_HOME
+      ${ANDROID_HOME}/tools/bin/sdkmanager --install "ndk;21.0.6113669"
+
+- task: Bash@3
+  displayName: Creation of emulator for testing
+  inputs:
+    targetType: inline
+    script: >-
+      #!/usr/bin/env bash
+
+      # Install AVD files
+      echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install 'system-images;android-27;google_apis;x86'
+
+      # Create emulator
+      echo "no" | $ANDROID_HOME/tools/bin/avdmanager create avd -n xamarin_android_emulator -k 'system-images;android-27;google_apis;x86' --force
```

Comment:
super nit: as this PR targets android and not xamarin we could rename the created virtual device to android_emulator

Reply:
Good idea, thanks!